src/web/llm-describe.tsx

```diff
@@ -0,0 +1,95 @@
+import {
+  createContext,
+  useContext,
+  useEffect,
+  useId,
+  type ReactNode,
+} from "react";
+
+export type LLMDescribeContent = string;
+
+export interface LLMDescribeNode {
+  id: string;
+  parentId: string | null;
+  content: string | null;
+}
+
+const nodes = new Map<string, LLMDescribeNode>();
+
+function setNode(node: LLMDescribeNode) {
+  nodes.set(node.id, node);
+  onChange();
+}
+
+function removeNode(id: string) {
+  nodes.delete(id);
+  onChange();
+}
+
+function onChange() {
+  const description = getLLMDescriptionString();
+  window.openai.setWidgetState({
+    ...window.openai.widgetState,
+    __widget_context: description,
+  });
+}
+
+const ParentIdContext = createContext<string | null>(null);
+
+interface LLMDescribeProps {
+  content: LLMDescribeContent | null | undefined;
+  children?: ReactNode;
+}
+
+export function LLMDescribe({ content, children }: LLMDescribeProps) {
+  const parentId = useContext(ParentIdContext);
+  const id = useId();
+
+  useEffect(() => {
+    setNode({
+      id,
+      parentId,
+      content: content ?? null,
```

Comment:
can you really set null on the content prop then? I feel like there is no benefit from tiggering `onChange` after registering a new node who's content is null.